cdap-proto/src/main/java/co/cask/cdap/proto/id/EntityId.java

```diff
@@ -132,7 +120,7 @@ public final String toString() {
     for (String part : idParts) {
       result.append(part).append(IDSTRING_PART_SEPARATOR);
     }
-    if (!Iterables.isEmpty(idParts)) {
+    if (idParts.iterator().hasNext()) {
```

Comment:
This can be avoided if you have the separator as the first char.

```java
StringBuilder builder = new StringBuilder(IDSTRING_TYPE_SEPARATOR);
for (String part : toIdParts()) {
  builder.append(part);
}
builder.subString(1);
```